benchmarks/src/main/java/org/elasticsearch/benchmark/xcontent/FilterContentBenchmark.java

```diff
@@ -0,0 +1,153 @@
+        Set<String> keys = flattenMap.keySet();
+        AtomicInteger count = new AtomicInteger();
+        switch (fieldCount) {
+            case "10_field":
+                return keys.stream().filter(key -> count.getAndIncrement() % 5 == 0).limit(10).collect(Collectors.toSet());
+            case "half_field":
+                return keys.stream().filter(key -> count.getAndIncrement() % 2 == 0).collect(Collectors.toSet());
+            case "all_field":
+                return new HashSet<>(keys);
+            case "wildcard_field":
+                return new HashSet<>(Arrays.asList("*stats"));
+            case "10_wildcard_field":
+                return Set.of(
+                    "*stats.nodes*",
+                    "*stats.ind*",
+                    "*sta*.shards",
+                    "*stats*.xpack",
+                    "*stats.*.segments",
+                    "*stat*.*.data*",
+                    inclusive ? "*stats.**.request_cache" : "*stats.*.request_cache",
+                    inclusive ? "*stats.**.stat" : "*stats.*.stat",
+                    inclusive ? "*stats.**.threads" : "*stats.*.threads",
+                    "*source_node.t*"
+                );
+            default:
+                throw new IllegalArgumentException("Unknown type [" + type + "]");
+        }
+    }
+
+    @Benchmark
+    public BytesReference filterWithParserConfigCreated() throws IOException {
+        return filter(this.parserConfig);
+    }
+
+    @Benchmark
+    public BytesReference filterWithNewParserConfig() throws IOException {
+        XContentParserConfiguration contentParserConfiguration = buildParseConfig();
+        return filter(contentParserConfiguration);
+    }
+
+    private XContentParserConfiguration buildParseConfig() {
+        Set<String> includes;
+        Set<String> excludes;
+        if (inclusive) {
+            includes = filters;
+            excludes = null;
+        } else {
+            includes = null;
+            excludes = filters;
+        }
+        return XContentParserConfiguration.EMPTY.withFiltering(includes, excludes);
+    }
+
+    private BytesReference filter(XContentParserConfiguration contentParserConfiguration) throws IOException {
+        try (BytesStreamOutput os = new BytesStreamOutput()) {
+            XContentBuilder builder = new XContentBuilder(XContentType.JSON.xContent(), os);
+            try (XContentParser parser = XContentType.JSON.xContent().createParser(contentParserConfiguration, source.streamInput())) {
+                if (parser.nextToken() != null) {
+                    builder.copyCurrentStructure(parser);
+                }
+                return BytesReference.bytes(builder);
+            }
+        }
+    }
+
+    private BytesReference readSource(String fileName) throws IOException {
+        return Streams.readFully(FilterContentBenchmark.class.getResourceAsStream(fileName));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getResourceAsStream*. Execution paths that do not contain closure statements were detected. Ensure the object returned by *getResourceAsStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).